crates/sui-transaction-builder/src/lib.rs

```diff
@@ -408,13 +408,19 @@ impl<Mode: ExecutionMode> TransactionBuilder<Mode> {
             package.linkage_table,
         )?;
 
+        // this value of max Move binary format version is used to deserialize a module for the
+        // purpose of obtaining types of for a Move call being constructed - using the max available
+        // format should be OK in this case
+        let max_binary_format_version =
+            ProtocolConfig::get_for_max_version().move_binary_format_version();
         let json_args_and_tokens = resolve_move_function_args(
             &package,
             module.clone(),
             function.clone(),
             type_args,
             json_args,
             Mode::allow_arbitrary_function_calls(),
+            max_binary_format_version,
```

Comment:
(suggestion in the comment above means you won't need to fudge this value, because you won't need to supply it any more).